File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithAutocreateComponent.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithAutocreateComponent {
    @Root
    default Class1 class1(Class2 class2) {
        return new Class1(class2);
    }


    record Class1(Class2 class2) {}

    final class Class2 {
        public Class2() {}
    }
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithAutocreateComponent.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithAutocreateComponent {
    @Root
    fun class1(class2: Class2): Class1 {
        return Class1(class2)
    }

    class Class1(private val class2: Class2)

    class Class2
}
